data/tutorials/language/1ms_02_dune.md

```diff
@@ -0,0 +1,241 @@
+```shell
+$ echo _build >> .gitignore
+```
+
+In OCaml, each source file is compiled into a module. In the `mixtli` project, the file `cloud.ml` creates a module named `Cloud`.
+
+Observe the roles of the different names:
+* `mixtli` is the project's name (it means *cloud* in Nahuatl).
+* `cloud.ml` is the OCaml source file's name, referred as `cloud` in the `dune` file.
+* `nube` is the executable command's name.
+* `Cloud` is the name of the module associated with the file `cloud.ml`.
+
+The `dune describe` command allows having a look at the project's module structure. Here is its output:
+```lisp
+((root /home/cuihtlauac/mixtli)
+ (build_context _build/default)
+ (executables
+  ((names (cloud))
+   (requires ())
+   (modules
+    (((name Cloud)
+      (impl (_build/default/cloud.ml))
+      (intf ())
+      (cmt (_build/default/.cloud.eobjs/byte/cloud.cmt))
+      (cmti ()))))
+   (include_dirs (_build/default/.cloud.eobjs/byte)))))
+```
+
+## Libraries
+<!-- CONCEPT: libraries -->
+
+When using Dune (with its default settings), an OCaml _library_ is a module aggregating other modules, bottom-up. This contrasts with the `struct ... end` syntax where modules are aggregated top-down by nesting submodules into container modules. Dune creates libraries from folders, like the following:
+```shell
+$ mkdir lib
+$ rm wmo.ml
+```
+
+**`lib/dune`**
+```lisp
+(library (name wmo))
+```
+
+All the modules found in the `lib` folder are bundled into the `Wmo` module.
+
+**`lib/cumulus.mli`**
+```ocaml
+val v : string
+val stratus : string
+```
+
+**`lib/cumulus.ml`**
+```ocaml
+let latin_root = "cumul"
+let v = latin_root ^ "us (Cu)"
+let stratus = "strato" ^ latin_root ^ "us (Sc)"
+```
+
+**`lib/stratus.mli`**
+```ocaml
+val v : string
+val cumulus : string
+```
+**`lib/stratus.ml`**
+```ocaml
+let latin_root = "strat"
+let v = latin_root ^ "us (St)"
+let cumulus = latin_root ^ "ocumulus (Sc)"
+```
+
+The executable and the corresponding `dune` file need to be updated to use the defined library as a dependency.
+
+**`dune`**
+```lisp
+(executable
+  (name cloud)
+  (public_name nube)
+  (libraries wmo))
+```
+
+
+
+**Observations**:
+* Dune creates a module `Wmo` from the contents of folder `lib`.
+* The folder's name (here `lib`) is irrelevant.
+* The library name appears uncapitalised (`wmo`) in `dune` files:
+  - In its definition, in `lib/dune`
+  - When used as a dependency in `dune`
+
+## Library Wrapper Modules
+<!-- CONCEPT: Submodule -->
+
+By default, modules bundled into a library by Dune are wrapped into a module. It is possible to bypass Dune's behaviour by manually writing the wrapper file.
+
+This `lib/wmo.ml` file corresponds to the module that Dune automatically generated in the previous section.
+
+**`lib/wmo.ml`**
+```ocaml
+module Cumulus = Cumulus
+module Stratus = Stratus
+```
+
+Here is how this makes sense of these module definitions:
```

Comment:
I don't understand this sentence. Might you mean:
- Here is how this makes sense in these module definitions 
or 
- Here is how to make sense of these module definitions
??